extensions/binding/org.eclipse.smarthome.binding.sonos/src/main/java/org/eclipse/smarthome/binding/sonos/handler/ZonePlayerHandler.java

```diff
@@ -586,34 +599,45 @@ private void updateGroupCoordinator() {
         }
     }
 
+    private boolean isUpnpDeviceRegistered() {
+        return service.isRegistered(this);
+    }
+
+    private void addSubscription() {
+        synchronized (upnpLock) {
+            // Set up GENA Subscriptions
+            if (service.isRegistered(this) && !subscriptionsRequested) {
+                for (String subscription : SERVICE_SUBSCRIPTIONS) {
+                    service.addSubscription(this, subscription, SUBSCRIPTION_DURATION);
+                }
+                subscriptionsRequested = true;
+            }
+        }
+    }
+
     private void removeSubscription() {
         synchronized (upnpLock) {
             // Set up GENA Subscriptions
             if (service.isRegistered(this)) {
                 for (String subscription : SERVICE_SUBSCRIPTIONS) {
                     service.removeSubscription(this, subscription);
                 }
-                service.unregisterParticipant(this);
             }
+            subscriptionsRequested = false;
+            service.unregisterParticipant(this);
         }
     }
 
     private void onUpdate() {
-        synchronized (upnpLock) {
-            for (String subscription : SERVICE_SUBSCRIPTIONS) {
-                service.addSubscription(this, subscription, SUBSCRIPTION_DURATION);
-            }
-
-            if (pollingJob == null || pollingJob.isCancelled()) {
-                Configuration config = getThing().getConfiguration();
-                // use default if not specified
-                int refreshInterval = DEFAULT_REFRESH_INTERVAL;
-                Object refreshConfig = config.get("refresh");
-                if (refreshConfig != null) {
-                    refreshInterval = ((BigDecimal) refreshConfig).intValue();
-                }
-                pollingJob = scheduler.scheduleAtFixedRate(pollingRunnable, 0, refreshInterval, TimeUnit.SECONDS);
+        if (pollingJob == null || pollingJob.isCancelled()) {
+            Configuration config = getThing().getConfiguration();
+            // use default if not specified
+            int refreshInterval = DEFAULT_REFRESH_INTERVAL;
+            Object refreshConfig = config.get("refresh");
+            if (refreshConfig != null) {
+                refreshInterval = ((BigDecimal) refreshConfig).intValue();
             }
+            pollingJob = scheduler.scheduleAtFixedRate(pollingRunnable, 0, refreshInterval, TimeUnit.SECONDS);
```

Comment:
I didn't mean to blame you for it, but as it was shown as a change in Github (although it was just moved code), I took the opportunity to add my remark :-)